orderfulfiller/orderfulfiller.go

```diff
@@ -106,25 +103,55 @@ func (r *OrderFulfiller) startOrderTimeoutWorker(ctx context.Context) {
 				lmt.Logger(ctx).Error("error getting expired orders", zap.Error(err))
 				continue
 			}
+
 			for _, order := range orders {
-				if fulfillmentStatus, err := r.fillHandler.UpdateFulfillmentStatus(ctx, order); err != nil {
+				fulfillmentStatus, err := r.fillHandler.UpdateFulfillmentStatus(ctx, order)
+				if err != nil {
 					lmt.Logger(ctx).Warn(
 						"error updating fulfillment status",
 						zap.Error(err),
 						zap.String("orderID", order.OrderID),
 						zap.String("sourceChainID", order.SourceChainID),
+						zap.String("destinationChainID", order.DestinationChainID),
 					)
-				} else if fulfillmentStatus == dbtypes.OrderStatusExpiredPendingRefund && r.shouldRefundOrders {
-					if err := r.fillHandler.InitiateTimeout(ctx, order); err != nil {
+					continue
+				}
+
+				// do not try and refund this order
+				if !r.shouldRefundOrders || fulfillmentStatus != dbtypes.OrderStatusExpiredPendingRefund {
+					continue
+				}
+
+				err = r.db.InTx(ctx, func(ctx context.Context, q db.Querier) error {
```

Comment:
Submitting the hyperlane transfer needs to be done in a db tx with the database insertion for the timeout. without this, if submitting the hyperlane transfer fails (this does a call on chain so it is possible for it to fail), then the hyperlane submit won't be retried and it won't be relayed.